internal/fields/mappings_test.go

```diff
@@ -0,0 +1,594 @@
+					"type": "keyword",
+				},
+			},
+			actual: map[string]any{
+				"bar": map[string]any{
+					"type": "keyword",
+				},
+				"metrics": map[string]any{
+					"type": "long",
+				},
+				"user": map[string]any{
+					"type": "text",
+				},
+			},
+			schema: []FieldDefinition{
+				{
+					Name:     "bar",
+					Type:     "keyword",
+					External: "ecs",
+				},
+				{
+					Name:     "metrics",
+					Type:     "keyword",
+					External: "ecs",
+				},
+				{
+					Name:     "user",
+					Type:     "keyword",
+					External: "",
+				},
+			},
+			expectedErrors: []string{
+				`field "user" is undefined: missing definition for path`,
+				`field "metrics" is undefined: actual mapping type (long) does not match with ECS definition type: keyword`,
+			},
+		},
+		{
+			title: "skip host group mappings",
+			preview: map[string]any{
+				"@timestamp": map[string]any{
+					"type": "keyword",
+				},
+				"host": map[string]any{
+					"properties": map[string]any{
+						"name": map[string]any{
+							"type": "text",
+						},
+					},
+				},
+			},
+			actual: map[string]any{
+				"@timestamp": map[string]any{
+					"type": "keyword",
+				},
+				"host": map[string]any{
+					"properties": map[string]any{
+						"os": map[string]any{
+							"type": "text",
+						},
+					},
+				},
+			},
+			schema:         []FieldDefinition{},
+			expectedErrors: []string{},
+		},
+		{
+			title: "missing mappings",
+			preview: map[string]any{
+				"@timestamp": map[string]any{
+					"type": "keyword",
+				},
+			},
+			actual: map[string]any{
+				"@timestamp": map[string]any{
+					"type": "keyword",
+				},
+				"foo": map[string]any{
+					"type": "keyword",
+				},
+			},
+			schema: []FieldDefinition{},
+			expectedErrors: []string{
+				`field "foo" is undefined: missing definition for path`,
+			},
+		},
+		{
+			title: "validate constant_keyword value",
+			preview: map[string]any{
+				"foo": map[string]any{
+					"type":  "constant_keyword",
+					"value": "example",
+				},
+			},
+			actual: map[string]any{
+				"foo": map[string]any{
+					"type":  "constant_keyword",
+					"value": "bar",
+				},
+			},
+			schema: []FieldDefinition{},
+			expectedErrors: []string{
+				`constant_keyword value in preview "example" does not match the actual mapping value "bar" for path: "foo"`,
+			},
+		},
+		{
+			title: "skip constant_keyword value",
+			preview: map[string]any{
+				"foo": map[string]any{
+					"type": "constant_keyword",
+				},
+			},
+			actual: map[string]any{
+				"foo": map[string]any{
+					"type":  "constant_keyword",
+					"value": "bar",
+				},
+			},
+			schema:         []FieldDefinition{},
+			expectedErrors: []string{},
+		},
+		{
+			title: "unexpected constant_keyword type",
+			preview: map[string]any{
+				"foo": map[string]any{
+					"type": "keyword",
+				},
+			},
+			actual: map[string]any{
+				"foo": map[string]any{
+					"type":  "constant_keyword",
+					"value": "bar",
+				},
```

Comment:
mmm it should not happen.
This test case was to ensure that this line is tested:
https://github.com/elastic/elastic-package/pull/2214/files#diff-67b718cebdeb840d99361c07166f75df079827a75df20850ef6cada861304521R467

```
	isConstantKeyword := false
	if mappingParameter("type", actual) != "constant_keyword" {
		return isConstantKeyword, nil
	}
	isConstantKeyword = true
	if mappingParameter("type", preview) != "constant_keyword" {
		return isConstantKeyword, fmt.Errorf("invalid type for %q: no constant_keyword type set in preview mapping", path)
	}
```

Would it be better to check first whether or not the type in the preview is `constant_keyword`? 
If the type is not constant_keyword in the preview, probably it could be returned nil too. WDYT @jsoriano ?

Reply:
Well, let's keep it as is just in case.